src/de/jost_net/JVerein/gui/action/SpendenbescheinigungSendAction.java

```diff
@@ -0,0 +1,109 @@
+* see <http://www.gnu.org/licenses/>.
+* 
+* https://openjverein.github.io
+**********************************************************************/
+package de.jost_net.JVerein.gui.action;
+
+import de.jost_net.JVerein.gui.view.SpendenbescheinigungMailView;
+
+import de.jost_net.JVerein.gui.dialogs.MailAbfrageDialog;
+import de.jost_net.JVerein.gui.dialogs.MailAbfrageDialog.Auswahl;
+import de.jost_net.JVerein.rmi.Spendenbescheinigung;
+import de.willuhn.jameica.gui.Action;
+import de.willuhn.jameica.gui.GUI;
+import de.willuhn.jameica.gui.parts.TablePart;
+import de.willuhn.jameica.system.OperationCanceledException;
+import de.willuhn.util.ApplicationException;
+
+/**
+ * E-Mail senden Formular anhand des zugewiesenen Spenders
+ */
+public class SpendenbescheinigungSendAction implements Action
+{
+  private de.willuhn.jameica.system.Settings settings;
+  
+  public SpendenbescheinigungSendAction()
+  {
+    super();
+    settings = new de.willuhn.jameica.system.Settings(this.getClass());
+    settings.setStoreWhenRead(true);
+  }
+
+  /**
+   * Versenden einer E-Mail mit der Spendenbescheinigung im Anhang 
+   * f�r Mitglieder deren Spendenbescheinigung im View ausgew�hlt ist.
+   */
+  @Override
+  public void handleAction(Object context) throws ApplicationException
+  {
+    Spendenbescheinigung[] spbArr = null;
+    // Pr�fung des Contexs, vorhanden, eine oder mehrere
+    if (context instanceof TablePart)
+    {
+      TablePart tp = (TablePart) context;
+      context = tp.getSelection();
+    }
+    if (context == null)
+    {
+      throw new ApplicationException("Keine Spendenbescheinigung ausgew�hlt");
+    }
+    else if (context instanceof Spendenbescheinigung)
+    {
+      spbArr = new Spendenbescheinigung[] { (Spendenbescheinigung) context };
+    }
+    else if (context instanceof Spendenbescheinigung[])
+    {
+      spbArr = (Spendenbescheinigung[]) context;
+    }
+    else
+    {
+      return;
+    }
+    
+    try
+    {
+      String text = "F�r das Versenden werden gedruckte Spendenbescheinigungen ben�tigt."
+          + "\nWenn sie schon gedruckt wurden brauchen sie nicht neu gedruckt werden.";
+      MailAbfrageDialog d = new MailAbfrageDialog(text, MailAbfrageDialog.POSITION_CENTER);
+      d.setTitle("Spendenbescheinigungen versenden");
+      d.setPanelText("Spendenbescheinigungen drucken?");
+      
+      MailAbfrageDialog.Auswahl choice = (MailAbfrageDialog.Auswahl) d.open();
+      if (choice == null)
+      {
+        return;
+      }
+      else if (choice == Auswahl.DRUCKEN_STANDARD)
+      {
+        SpendenbescheinigungPrintAction action = new SpendenbescheinigungPrintAction(true, true);
+        action.handleAction(spbArr);
+      }
+      else if (choice == Auswahl.DRUCKEN_INDIVIDUELL)
+      {
+        SpendenbescheinigungPrintAction action = new SpendenbescheinigungPrintAction(false, true);
+        action.handleAction(spbArr);
+      }
+
```

Comment:
in MitgliedMenu.java:95 wird vor dem Erstellen des Mails eine Vorlage zur Auswahl angeboten. Ich hab mir das nicht angeschaut und weiß auch nicht, ob das einfach einzubauen ist, aber mir würde diese Funktion schon helfen. Wenn das nicht geht, ist es auch nicht schlimm.

Reply:
Dein Vorschlag mit den Vorlagen, dass man die mit einem Button einblenden und dann auswählen kann, ist super! Es ist schon ein bisschen verwirrend, wenn zuerst immer der Vorlagen-Dialog angezeigt wird. Wenn man sich dran gewöhnt hat, ist das ja kein Problem mehr, aber es ist halt nicht intuitiv. Da ist Dein Vorschlag schon sehr viel besser!